### PR TITLE
Add option for Mapbox API key

### PR DESCRIPTION
**Pull request by marecabo**
# Proposed Changes

This change adds an option to specify the Mapbox API key to enable the map view on a tag's page.

Tested on RPi3B+ 32bit.

## Related Issues

https://github.com/hassio-addons/addon-firefly-iii/issues/8

[autolink-references]: https://help.github.com/articles/autolinked-references-and-urls/

### CODE REVIEW

File: firefly-iii/config.json [hunk 1]
```diff
@@ -32,6 +33,7 @@
     "log_level": "list(trace|debug|info|notice|warning|error|fatal)?",
     "certfile": "str",
     "keyfile": "str",
-    "ssl": "bool"
+    "ssl": "bool",
+    "mapbox_api_key": "str"
```
Comment (sinclairpaul) [inline firefly-iii/config.json]: Should be optional "str?"

File: firefly-iii/config.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
   "options": {
     "ssl": false,
     "certfile": "fullchain.pem",
-    "keyfile": "privkey.pem"
+    "keyfile": "privkey.pem",
+    "mapbox_api_key": ""
```
Comment (sinclairpaul) [inline firefly-iii/config.json]: We can remove it here if its optional

File: firefly-iii/rootfs/etc/services.d/php-fpm/run [hunk 3]
```diff
@@ -4,10 +4,13 @@
 # Runs the PHP-FPM daemon
 # ==============================================================================
 
+CONFIG_PATH=/data/options.json
+
 declare key
 
 key=$(cat /data/firefly/appkey.txt)
 export APP_KEY=${key}
+export MAPBOX_API_KEY=$(jq --raw-output ".mapbox_api_key" $CONFIG_PATH)
```
Comment (sinclairpaul) [inline firefly-iii/rootfs/etc/services.d/php-fpm/run]: We can use a bashio method, you can see examples in the other scripts

File: firefly-iii/rootfs/etc/services.d/php-fpm/run [hunk 4]
```diff
@@ -4,10 +4,13 @@
 # Runs the PHP-FPM daemon
 # ==============================================================================
 
+CONFIG_PATH=/data/options.json
```
Comment (sinclairpaul) [inline firefly-iii/rootfs/etc/services.d/php-fpm/run]: No need for this

**Review by sinclairpaul — CHANGES_REQUESTED** [review]

**Comment by marecabo** [comment]
Thank you for your comments. I will incorporate them asap.

**Comment by marecabo** [comment]
I just revised the changes with your recommendations and also tested the addon again with and without the `mapbox_api_key` option set.

**Review by sinclairpaul — APPROVED** [review]

**Comment by sinclairpaul** [comment]
Thanks @marecabo

**Comment by marecabo** [comment]
I just saw the failing check here: https://gitlab.com/hassio-addons/addon-firefly-iii/-/jobs/584302282

Sorry for that, I did not expect it to fail there as the `APP_DEBUG` variables are also not declared previously.

Do you want me to just move the empty declaration of the variable `MAPBOX_API_KEY` before the if branch as it is done with the others?

<blockquote><div>GitLab</div><div><strong><a href="https://gitlab.com/hassio-addons/addon-firefly-iii/-/jobs/584302282">shellcheck (#584302282) · Jobs · Community Hass.io Add-ons / addon-firefly-iii</a></strong></div><div>Firefly-III - Community Hass.io Add-on for Home Assistant</div></blockquote>

**Comment by sinclairpaul** [comment]
No problem, I've fixed it, I should of spotted it anyway 🤦

**Pull request merged**